Implement fetching user's active orders query in SQS

deividaspetraitis opened the pull request:
This PR introduces a new `GetActiveOrders` method for fetching active orders by owner from the orderbook contract. 

Additionally refactors response returned by `GetTickUnrealizedCancels` and adds `OrderbookGRPCClientMock`.

deividaspetraitis committed:
SQS-344 | Implement fetching user's active orders query

Introduces a new GetActiveOrders method for fetching active orders by owner from the orderbook contract.
Additionally refactors response returned by GetTickUnrealizedCancels and adds OrderbookGRPCClientMock.

p0mvn commented on domain/orderbook/plugin/orderbook_tick.go:
```diff
@@ -0,0 +1,9 @@
+package orderbookplugindomain
+
+import orderbookdomain "github.com/osmosis-labs/sqs/domain/orderbook"
+
+// UnrealizedTickCancels is a struct that represents the response payload from orderbook for unnrealized cancels by tick.
+type UnrealizedTickCancels struct {
+	TickID                 int64                             `json:"tick_id"`
+	UnrealizedCancelsState orderbookdomain.UnrealizedCancels `json:"unrealized_cancels"`
+}
```
What was the motivation for moving it here?

`orderbookplugindomain` is for anything relevant to the fill bot (plugin). The fill bot does not use this type so the original package `orderbookgrpcclientdomain` might be a better fit because it is more general.

deividaspetraitis replied:
I have refactored because `GetTickUnrealizedCancels` returned unexported data type making imposible to implement for Mock for the client. As a result I made it consistent with other metods as well, instead just by making old data structure exported.

p0mvn replied:
I see - that makes sense but we should still be able to move this back to `orderbookgrpcclientdomain` it sounds like. Could we please do it?

deividaspetraitis replied:
I am still feel confused, all other [OrderBookClient](https://github.com/osmosis-labs/sqs/blob/130a34edb41f978f649cbdc0d7ad5ef7dd01616d/domain/orderbook/grpcclient/orderbook_grpc_client.go#L13) methods uses data structures from `orderbookplugindomain`, and this package has its own gRPC clients, such as [OrderbookCWAPIClient](https://github.com/osmosis-labs/sqs/blob/130a34edb41f978f649cbdc0d7ad5ef7dd01616d/domain/orderbook/plugin/orderbook_grpc_client.go#L8). Addressed request with [130a34e](https://github.com/osmosis-labs/sqs/pull/459/commits/130a34edb41f978f649cbdc0d7ad5ef7dd01616d), merging this PR to unblock us moving forward. :rocket:

p0mvn reviewed (approved):
LGTM - nice work - only one nit

deividaspetraitis committed:
SQS-344 | Implement fetching user's active orders query

Requested change: Move UnrealizedTickCancels back to orderbookgrpcclientdomain package
